include/ck/utility/amd_gemm_dpp.hpp

```diff
@@ -11,33 +11,57 @@ namespace ck {
 
 namespace dpp8 {
 
-template <index_t MPerLanegroup,
-          index_t NPerLanegroup,
-          index_t KPerLanegroup,
-          class FloatA,
-          class FloatB,
-          class FloatVecC,
+template <class ABDataType>
+struct dpp_datatypes;
+
+template <>
+struct dpp_datatypes<half_t>
+{
+    // Dot product of `half2_t` and `half2_t` to get `float`. Reducing 2 elements from K in a
+    // single instruction.
+    using a_dtype                        = half_t;
+    using b_dtype                        = half_t;
+    using c_dtype                        = float;
+    static constexpr index_t k_per_instr = 2;
+};
+
+template <index_t MPerThread,
+          index_t NPerThread,
+          index_t KPerThread,
+          class BaseInputType,
+          class AVecDataType,
+          class BVecDataType,
+          class CVecDataType,
           bool ShareA>
-__device__ void RunGemm(const FloatA& a, const FloatB& b, FloatVecC& c_vec)
+struct DppInstrRunner
 {
-    constexpr index_t c_dim = ShareA ? MPerLanegroup : NPerLanegroup;
-
-    const vector_type<half_t, KPerLanegroup> a_vector{a};
-    const vector_type<half_t, KPerLanegroup> b_vector{b};
-
-    static_for<0, c_dim, 1>{}([&](auto c_idx) {
-        float c = c_vec.template AsType<float>()(c_idx);
-        // Next `c_idx` implies that we need to pull data from the next lane.
-        constexpr index_t source_lane = c_idx;
-        static_for<0, KPerLanegroup / 2, 1>{}([&](auto k_chunk) {
-            const auto a_half2 = a_vector.template AsType<half2_t>()[k_chunk];
-            const auto b_half2 = b_vector.template AsType<half2_t>()[k_chunk];
-            ck::dpp8::inner_product_dpp<half2_t, half2_t, float, source_lane, ShareA>(
-                a_half2, b_half2, c);
+    static constexpr auto datatypes_conf = dpp_datatypes<BaseInputType>{};
+    using ADataType                      = typename decltype(datatypes_conf)::a_dtype;
+    using BDataType                      = typename decltype(datatypes_conf)::b_dtype;
+    using CDataType                      = typename decltype(datatypes_conf)::c_dtype;
```

Comment:
Couldn't it work like that: 
```suggestion
    using datatypes_conf = dpp_datatypes<BaseInputType>;
    using ADataType                      = typename datatypes_conf::a_dtype;
    using BDataType                      = typename datatypes_conf::b_dtype;
    using CDataType                      = typename datatypes_conf::c_dtype;
```

Reply:
Sure, that's a great point! Changed it.